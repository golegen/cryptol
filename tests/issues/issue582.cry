foo : {i, j}[i /^ j]
foo = zero

bar : {i, j}[i %^ j]
bar = zero
