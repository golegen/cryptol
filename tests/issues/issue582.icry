zero : [1 /^ inf]
zero : [inf /^ 1]
zero : [1 /^ 0]
zero : [1 %^ inf]
zero : [inf %^ 1]
zero : [1 %^ 0]
:l issue582.cry
