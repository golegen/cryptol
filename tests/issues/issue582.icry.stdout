Loading module Cryptol

[error] at <interactive>:1:1--1:18:
  Expected a finite type, but found `inf`.

[error] at <interactive>:1:1--1:18:
  Expected a finite type, but found `inf`.

[error] at <interactive>:1:1--1:16:
  Unsolvable constraint: 0 >= 1

[error] at <interactive>:1:1--1:18:
  Expected a finite type, but found `inf`.

[error] at <interactive>:1:1--1:18:
  Expected a finite type, but found `inf`.

[error] at <interactive>:1:1--1:16:
  Unsolvable constraint: 0 >= 1
Loading module Cryptol
Loading module Main

[error] at ./issue582.cry:2:1--2:11:
  Failed to validate user-specified signature.
    in the definition of 'Main::foo', at ./issue582.cry:2:1--2:4,
    we need to show that
      for any type i, j
      the following constraints hold:
        • fin j
            arising from
            use of partial type function (/^)
            at ./issue582.cry:1:7--1:21
        • fin i
            arising from
            use of partial type function (/^)
            at ./issue582.cry:1:7--1:21
        • j >= 1
            arising from
            use of partial type function (/^)
            at ./issue582.cry:1:7--1:21
[error] at ./issue582.cry:5:1--5:11:
  Failed to validate user-specified signature.
    in the definition of 'Main::bar', at ./issue582.cry:5:1--5:4,
    we need to show that
      for any type i, j
      the following constraints hold:
        • fin j
            arising from
            use of partial type function (%^)
            at ./issue582.cry:4:7--4:21
        • fin i
            arising from
            use of partial type function (%^)
            at ./issue582.cry:4:7--4:21
        • j >= 1
            arising from
            use of partial type function (%^)
            at ./issue582.cry:4:7--4:21
